libdnf/hy-iutil.cpp

```diff
@@ -459,6 +459,27 @@ pool_tmpdup(Pool *pool, const char *s)
     return strcpy(dup, s);
 }
 
+static Id
+running_kernel_check_path(DnfSack *sack, const char *fn)
+{
+    if (access(fn, F_OK))
+        g_debug("running_kernel_check_path(): no matching file: %s.", fn);
+
+    HyQuery q = hy_query_create_flags(sack, HY_IGNORE_EXCLUDES);
+    dnf_sack_make_provides_ready(sack);
+    hy_query_filter(q, HY_PKG_FILE, HY_EQ, fn);
+    hy_query_filter(q, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
```

Comment:
Please could you replace following lines
```
    hy_query_filter(q, HY_PKG_FILE, HY_EQ, fn);
    hy_query_filter(q, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
```
by
```
    q->installed()
    hy_query_filter(q, HY_PKG_FILE, HY_EQ, fn);
```
`installed()` is about 10x faster then `hy_query_filter(q, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);`.
Using `installed()` before `hy_query_filter(q, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);` also gain a lot of performance improvement.

Reply:
Done.